src/pke/lib/scheme/bfvrns/bfvrns-cryptoparameters.cpp

```diff
@@ -684,16 +684,15 @@ void CryptoParametersBFVRNS::PrecomputeCRTTables(KeySwitchTechnique ksTech, Scal
         auto tmp = m_msk;
         while (tmp > 0) {
             tmp >>= 1;
-            s++;
+            ++s;
         }
 
         BigInteger Q(GetElementParams()->GetModulus());
         BigInteger maxConvolutionValue(BigInteger(2 * n) * BigInteger(GetPlaintextModulus()) * Q * Q);
         // check msk is large enough
         while (Q * B * BigInteger(m_msk) < maxConvolutionValue) {
-            auto firstInteger{FirstPrime<NativeInteger>(s + 1, 2 * n)};
-            m_msk = NextPrime<NativeInteger>(firstInteger, 2 * n);
-            if (++s >= 60)
+            m_msk = LastPrime<NativeInteger>(++s, 2 * n);
+            if (s >= 60)  // TODO: MAX_MODULUS_SIZE?
```

Comment:
I think this check should be done before calling `LastPrime`

Reply:
Should I remove the check completely?
For NativeInteger, there is a check to see if ++s is greater than MAX_MODULUS_SIZE within the LastPrime call. Behavior for NATIVESIZE=64 would be exactly the same, but s could potentially grow larger than 60 for NATIVESIZE=128.

Reply:
Yes, you can remove it